build_scripts/build_usd.py

```diff
@@ -201,14 +201,29 @@ def RunCMake(context, force, extraArgs = None):
 
     generator = context.cmakeGenerator
 
+    # OpenSubdiv seems to error when building on windows w/ Ninja...
```

Comment:
I'd prefer to have library-specific behavior set up in that library's Install... function, rather than putting it in RunCMake. That way, all of the quirks for building a specific library are contained in a single place for ease of inspection.

So, would it be possible to move this logic into InstallOpenSubdiv instead? I was thinking you could do this check in that function and if it's required, then temporarily set context.cmakeGenerator to None, call RunCMake, then reset it back to its original value (probably in a try-finally block).

Reply:
If we did this, you could revert the changes that add currentDependency to the context object as well as the above change.

Reply:
Done and done in latest changes!